solutions/cfgym-102951C.mdx

```diff
@@ -74,4 +74,56 @@ int main() {
 
 </CPPSection>
 
+<JavaSection>
+	
+``` java
+import java.io.*;
+
+import java.util.*;
+
+
+public class Main {
+	//CodeSnip{Kattio}
+	public static void main(String[] args) {
+		Kattio io = new Kattio();
+
+		int n = io.nextInt();		
+		int[] a = new int[n+1], b = new int[n+1], c = new int[n+1], lis = new int[n+1];
+		int[] pos = new int[n+2];
+
+		for(int i = 1;i<=n;i++)
+		{
+			a[i] = io.nextInt();
+			 //Pos is the inverse of a
+			pos[a[i]] = i;
+		}
+
+		for(int i = 1;i<=n;i++)
+		{
+			b[i] = io.nextInt();
+		}
+
+		for(int i = 1;i<=n;i++)
+		{
+			c[i] = pos[b[i]];
+		}
+
+		//Length of the LIS
+		int len = 0; 
+		for(int i = 1;i<=n;i++)
+		{
+			//Finding LowerBound
+			int p = Arrays.binarySearch(lis,0,len,c[i]); 
+			if(p < 0) p = Math.abs(p + 1);
+			lis[p] = c[i];
+			if(p == len) len++;
+		}
+		
+		System.out.println(len);
+	}
```

Comment:
close the kattio by doing io.close();

nothing will print if you don't do this

Reply:
My bad. Will commit adding it in.